Dockerfile

```diff
@@ -1,14 +1,57 @@
 # https://hub.docker.com/r/jenkins/jenkins/tags/
-FROM jenkins/jenkins:2.102-alpine
+FROM jenkins/jenkins:2.104-alpine
 
+# set maintainer
+LABEL maintainer "@cloudposse"
+
+# change user
 USER root
 
-RUN apk update && apk upgrade && \
-    apk add --no-cache bash git openssh gettext make docker
+# install required packages
+RUN readonly PACKAGES=" \
```

Comment:
@ivan-pinatti what are your thoughts on converting this instead to an `ARG`?

Reply:
@osterman I did it for testing purposes on my environment, however, AWS CodeBuild is running an older version of Docker that doesn't support it. So, for now we can't change.

I already submitted a PR to AWS CodeBuild adding a newer Docker Image version of [Ubuntu](https://github.com/aws/aws-codebuild-docker-images/pull/27) and on top of that a newer version of [Docker](https://github.com/aws/aws-codebuild-docker-images/pull/28).

The problem is that I submitted both more than a month ago and I did not receive any interaction from AWS, it seems that they are not much attentive to the repo.

Reply:
Oh wow, that's crazy! Feels like this feature has been in docker for ages now.

Reply:
Oh, do you mean that they don't support multi-line `ARG`? I see you're using it everywhere else. Could you add a link to your PR?

Reply:
@osterman I was referring to the Jenkins version itself, in the first lines of the Dockerfile, ex:

```
# set default version if no argument was provided
ARG JENKINS_VERSION="2.103"
FROM jenkins/jenkins:${JENKINS_VERSION}-slim
```

This is not supported yet.

Regarding to pass the package list as argument I don't see any gain on doing it at the moment.

Reply:
> Regarding to pass the package list as argument I don't see any gain on doing it at the moment.

personal bias is in favor of reducing the inline bash, when the dockerfile language defines a convention for this.

Reply:
You mean, you prefer this:

```
# install required packages
RUN apk update && apk upgrade && apk add --no-cache bash=4.4.12-r2 docker=17.10.0-r0 gettext=0.19.8.1-r1 git=2.15.0-r1 make=4.2.1-r0 openssh=7.5_p1-r8 openssl=1.0.2n-r0
```

Instead of this:

```
# install required packages
RUN readonly PACKAGES=" \
      bash=4.4.12-r2 \
      docker=17.10.0-r0 \
      gettext=0.19.8.1-r1 \
      git=2.15.0-r1 \
      make=4.2.1-r0 \
      openssh=7.5_p1-r8 \
      openssl=1.0.2n-r0 \
    " \
    && apk update \
    && apk upgrade \
    && apk add --no-cache ${PACKAGES}
```

?

Reply:
More like:
```
ARG PACKAGES="bash=4.4.12-r2 docker=17.10.0-r0 gettext=0.19.8.1-r1 git=2.15.0-r1 make=4.2.1-r0 openssh=7.5_p1-r8 openssl=1.0.2n-r0"
RUN apk update && apk upgrade && apk add --no-cache ${PACKAGES} ...
```

Reply:
Considering this as integrated part of Terraform Jenkins AWS, we would have to add it as a parameter into AWS CodeBuild code section as well, and I see no advantage in doing it. At least not with the current solution design.

Even though, I would prefer to have it as multi-line for better readability and improved code maintenance, something like:
```
# install required packages
ARG PACKAGES=" \
      bash=4.4.12-r2 \
      docker=17.10.0-r0 \
      gettext=0.19.8.1-r1 \
      git=2.15.0-r1 \
      make=4.2.1-r0 \
      openssh=7.5_p1-r8 \
      openssl=1.0.2n-r0 \
    " \

RUN apk update \
    && apk upgrade \
    && apk add --no-cache ${PACKAGES}
```

However, I'm not sure that Docker supports ARG multi-line syntax yet, as seen here:
(https://github.com/moby/moby/issues/35950)[https://github.com/moby/moby/issues/35950]